src/openssl/lib_crypto.cr

```diff
@@ -1,11 +1,23 @@
 {% begin %}
   lib LibCrypto
-    OPENSSL_110 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.1.0 libcrypto || printf %s false`.stringify != "false" }}
-    OPENSSL_102 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.0.2 libcrypto || printf %s false`.stringify != "false" }}
+    # An extra zero is appended to the output of LIBRESSL_VERSION to make it 0 when LibreSSL does not exist on the system.
+    # Any comparisons to it should be affixed with an extra zero as well e.g. `(LIBRESSL_VERSION_NUMBER >= 0x2050500F0)`.
+    LIBRESSL_VERSION = {{ system("echo \"#include <openssl/opensslv.h>\nLIBRESSL_VERSION_NUMBER\" | " +
+                                 (env("CC") || "cc") + " " + `command -v pkg-config > /dev/null && pkg-config --cflags --silence-errors libssl || true`.chomp.stringify + " -E -").chomp.split('\n').last.split('L').first.id + "0" }}
```

Comment:
Why do we need `command -v` *and* `|| true`?

Reply:
Hum, the `pkg-config` call can fail so we need the `|| true`, so I guess we don't need the `command -v` anymore (just try calling `pkg-config` and let if fail).

Reply:
`command -v` produces no output on error while `command` does. `|| true` is so the shellstring returns empty string on error. We need both as we need to work with environments with no `pkg-config` and environments with `pkg-config` but no `.pc` files for openssl.

Reply:
Oh wait `|| true` can cover both cases, right.

Reply:
Oh wait `|| true` can cover both cases, right.